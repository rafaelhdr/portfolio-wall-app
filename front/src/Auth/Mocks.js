
const EXPECTED_BODY = {

  REGISTER_SUCCESSFUL: {
    errors: null,
  }

}

export { EXPECTED_BODY }
