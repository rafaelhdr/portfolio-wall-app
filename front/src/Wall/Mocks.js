
const EXPECTED_BODY = {

  LIST_OF_POSTS: {
    errors: null,
    data: {
      posts: [
        {
          id: '1',
          message: 'War is peace. Freedom is slavery. Ignorance is strength.',
          author_name: 'George Orwell',
          created_at: '2018-05-05T23:45:59.733621Z',
        },
        {
          id: '2',
          message: 'All animals are equal, but some animals are more equal than others.',
          author_name: 'George Orwell',
          created_at: '2018-05-05T22:23:00.065067Z',
        },
      ]
    }
  },

  NEW_POST_SUCCESSFUL: {
    errors: null,
  },

}

class MockApp {
  constructor() {
    this.setUserCalled = false;
    this.updatePostsCalled = false;
  }
  setUser(user) {
    this.setUserCalled = true;
  }
  updatePosts() {
    this.updatePostsCalled = true;
  }
}

export { EXPECTED_BODY, MockApp }
