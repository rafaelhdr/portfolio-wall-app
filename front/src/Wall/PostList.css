p.author_line {
    font-size: small;
}
