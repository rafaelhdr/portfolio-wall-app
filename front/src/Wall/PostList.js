import React, { Component } from 'react';
import './PostList.css';


function humanizeDate(inputDate) {
  const eventDate = new Date(Date.parse(inputDate));
  return eventDate.toUTCString();
}

class PostList extends Component {

  render() {
    if (this.props.posts === null) {
      return <h3>Loading messages</h3>
    }

    return (
      <div>
        <div className="row">
          <div className="col-sm-12">
            <h2>List of messages</h2>
          </div>
        </div>

        {this.props.posts.map(({ id, message, author_name, created_at }) => {
          return (
            <div key={id} className="jumbotron">
              <p>{message}</p>
              <p className="author_line">{author_name} - {humanizeDate(created_at)}</p>
            </div>
          )
        })}

      </div>
    )
  }
}

export { PostList }
