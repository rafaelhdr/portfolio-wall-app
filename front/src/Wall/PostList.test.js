import React from 'react';
import { PostList } from "./PostList.js";
import renderer from 'react-test-renderer';
import { EXPECTED_BODY, MockApp } from './Mocks.js';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });
import moxios from 'moxios';


describe('API', () => {

  describe('methods', () => {

    it('list posts of the wall', () => {
      const posts = EXPECTED_BODY.LIST_OF_POSTS.data.posts;
      const tree = renderer
        .create(<PostList posts={posts} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

  })

});
