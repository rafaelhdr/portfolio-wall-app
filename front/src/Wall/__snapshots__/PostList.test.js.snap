// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API methods list posts of the wall 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="col-sm-12"
    >
      <h2>
        List of messages
      </h2>
    </div>
  </div>
  <div
    className="jumbotron"
  >
    <p>
      War is peace. Freedom is slavery. Ignorance is strength.
    </p>
    <p
      className="author_line"
    >
      George Orwell
       - 
      Sat, 05 May 2018 23:45:59 GMT
    </p>
  </div>
  <div
    className="jumbotron"
  >
    <p>
      All animals are equal, but some animals are more equal than others.
    </p>
    <p
      className="author_line"
    >
      George Orwell
       - 
      Sat, 05 May 2018 22:23:00 GMT
    </p>
  </div>
</div>
`;
